var/spack/repos/builtin/packages/py-poetry-core/package.py

```diff
@@ -10,17 +10,25 @@ class PyPoetryCore(PythonPackage):
     """Poetry PEP 517 Build Backend."""
 
     homepage = "https://github.com/python-poetry/poetry-core"
-    pypi = "poetry-core/poetry-core-1.0.7.tar.gz"
+    pypi = "poetry_core/poetry_core-1.6.1.tar.gz"
```

Comment:
Both dash and underscore work for the first part, it's only the second part where it matters. I think dash is technically more correct since that's the package name on PyPI. So maybe:
```suggestion
    pypi = "poetry-core/poetry_core-1.6.1.tar.gz"
```